### DISCUSSION TITLE
Feedback from user testing

### DISCUSSION THREAD

**Discussion by JPReceveur** [IGS/gEAR#168] [Ideas]
The following comments are a condensed list of items identified during user testing to turn into issues/enhancements. 
I can seperate these into tool/ feature specific lists but just want to have a general record of feedback in one spot first.

**Comment by JPReceveur** [comment]
When adding datasets to a new profile created in the dataset explorer window, if you click on the view datasets in profile option, it indicates that 220 datasets are in the current profile (in the example I only added 4)

Replicated bug by:
1. Opening dataset explorer and creating a new profile named test
2. Add 4 random datasets to profile with the + button
3. Clicked on view datasets in profile

<img width="466" alt="Screen Shot 2022-01-19 at 9 43 03 AM" src="https://user-images.githubusercontent.com/28451557/150153356-8c0a4cec-6ddc-4a14-8a3a-e57648ba433a.png">

After you click off/ refresh the page, the profiles work as expected (only showing the profiles that were added), but seeing the 220 in profile when they only selected 4 tripped up a couple of the test users.


<img width="1369" alt="Screen Shot 2022-01-19 at 9 44 55 AM" src="https://user-images.githubusercontent.com/28451557/150153792-d6747ce1-85ad-4a74-94f3-587252379823.png">

**Comment by JPReceveur** [comment]
Two of the users had issues logging in (remembering their password) and requested a way to reset/recover site password.

On a somewhat related note, ended up on this page by accidentally clicking on create an account and then signing in while on that page
<img width="1275" alt="Screen Shot 2022-01-19 at 10 06 03 AM" src="https://user-images.githubusercontent.com/28451557/150157765-51d1a0e7-7937-4aad-ab23-07dbaecb1e69.png">

**Reply by JPReceveur** [reply]
Tested this out by putting in a wrong password and realized the functionality already exists. Should the forget password option be available by default when signing in and then get hidden after login to keep the page less cluttered? Users might not know to put in a wrong password in order to get the forget password option to come up.

**Comment by JPReceveur** [comment]
Easier way to get to comparison/analysis tools from dataset explorer:

When trying to use tools (either single-cell or comparison) on a particular dataset, most of the users searched for the dataset in the dataset explorer and then wanted to open it in a particular tool.
When using the search by gene expression, users can open up the dataset in an analysis tool easily with the dropdown menu but to open the dataset in a tool from the dataset explorer users have to click on view > search for a gene > then use the dropdown menu to open dataset in one of the tools.

Maybe add buttons to the options bar in the dataset explorer so they can go directly from dataset to the analysis tool?

 <img width="702" alt="Screen Shot 2022-01-19 at 10 14 20 AM" src="https://user-images.githubusercontent.com/28451557/150160056-5360131f-6961-45de-9a74-19a8dadc94b2.png">

**Comment by JPReceveur** [comment]
Ambiguity between compare and analyze in dropdown:

In the dropdown menu for getting more information on datasets, compare takes you to the comparison tool while analyze takes you to the single cell tool. Could the names of these be changed to better reflect the tool the link is taking you to? 
Maybe 'Compare Gene Expression' and 'Analyze Single Cell'?


<img width="414" alt="Screen Shot 2022-01-19 at 10 21 03 AM" src="https://user-images.githubusercontent.com/28451557/150160786-71809c85-72d1-47c3-baaa-8826f560ba75.png">

**Reply by adkinsrs** [reply]
Addressed this in 851ca5b

Should be in gear-devel

**Reply by JPReceveur** [reply]
Looks good on devel

**Comment by JPReceveur** [comment]
Users unclear on steps to move forward on single cell workbench:

In the single-cell workbench, after users select a dataset and run the first filtering step, they were unclear on that they had to click the next toggle to move on. Perhaps naming the steps in order (1,2,3,etc) or including some text that pops up after they apply filters that they need to click the next toggle would be useful.

<img width="498" alt="Screen Shot 2022-01-19 at 10 28 00 AM" src="https://user-images.githubusercontent.com/28451557/150162574-b2eafd9a-f317-479a-93ba-4ba985cf0c6e.png">

**Reply by adkinsrs** [reply]
I have gone ahead and added some prompts to click the next toggle after each step completes. When Joshua updates the production codebase, these should be available.  Any feedback would be appreciated (when the update happens).

**Reply by adkinsrs** [reply]
And... it's updated.

**Reply by JPReceveur** [reply]
Just went through and the extra prompts look good. 

Random observation: when choosing 500 top genes, the number at the top shows 499 (dataset = 5dpf zebrafish, Piotrowski)

 <img width="670" alt="Screen Shot 2022-01-31 at 2 55 07 PM" src="https://user-images.githubusercontent.com/28451557/151864937-bce36011-c90b-4993-bd94-61405ee5fbd2.png">

**Reply by adkinsrs** [reply]
I did 500 on a dataset and the number in parenthesis was 500, so there is definitely some processing to get to that final number.  Will have to look into that as well.

**Reply by adkinsrs** [reply]
Wow this is a "throw your hands up and call it a day" issue.  If I test with any other number (100, 300, 499, 501, 3333) I get that number in the parenthesis.  But if I use 500, it shows 499 which is being determined by the shape of the h5ad expression matrix after identifying highly variable genes. And I have only encountered this on this particular dataset, though I am sure other datasets have a similar issue if I cherry-picked numbers to test like I did this one.

**Reply by adkinsrs** [reply]
Ran this with 500 top genes (which returns 499) and 501 top genes, and here are the two genes that are in the latter dataset

```
                   gene_symbol  n_cells  highly_variable     means  dispersions  dispersions_norm
index
ENSDARG00000027572       as3mt       56             True  0.250963     2.318369          1.807868
ENSDARG00000061268        ago2        9             True  0.052605     2.515744          1.807624
```

Since the cause seems to be hard to determine at the moment, I'm going to drop this for now and resume later on.

**Comment by JPReceveur** [comment]
Overplotting in comparison tool plot:
In the comparison tool, the plot controls and title tend to be overplotted, which makes the plot controls hard to see/interact with.


<img width="1085" alt="Screen Shot 2022-01-19 at 10 35 55 AM" src="https://user-images.githubusercontent.com/28451557/150164141-9bb0c71c-505d-4660-9f0e-45b89711a7ed.png">

**Reply by adkinsrs** [reply]
This one is always going to be touchy given that the title is the name of the dataset.  I could simply move the title into a <div> above the plot, but then it would be excluded if one were to download the plot.  And I don't think reducing the font size will fix a number of these. I will ponder this one over.... I think I can rotate the plot controls vertically

**Reply by adkinsrs** [reply]
One option I found is that I can set the plot to allow editing of various annotations and labels with a click, such as the title text.  It's not an intuitive fix since the user would probably need to read the instructions to know it is there, but it is an option.

**Reply by adkinsrs** [reply]
<img width="1728" alt="Screen Shot 2022-02-02 at 4 06 29 PM" src="https://user-images.githubusercontent.com/5665914/152237359-271a914a-34bc-4969-9219-479a69a954ee.png">

Edit with modebar in vertical orientation instead of horizontal.  I am thinking this is the way to go.

**Reply by JPReceveur** [reply]
I agree with the vertical orientation and it puts the plot controls closer to the instructions for them. Maybe just change the wording in the Plot controls from 'upper right' to 'to the right of the plot'?

**Reply by adkinsrs** [reply]
Done... pushed a commit for this.

**Comment by JPReceveur** [comment]
Info button on single cell wb doesn't open:

In the single cell workbench, the info button on the QC by mitochondrial content step doesn't pop up with any information

<img width="398" alt="Screen Shot 2022-01-19 at 10 45 28 AM" src="https://user-images.githubusercontent.com/28451557/150165354-163d1a0b-b2cc-43fa-9cfa-7783a3df43fe.png">

**Reply by JPReceveur** [reply]
In current testing on devel, I was able to the info box to show up by reloading and clicking around on other items first. Will take another look into when it doesn't show up later.

Also breakpoints were showing up in the text

<img width="597" alt="Screen Shot 2022-01-19 at 11 00 26 AM" src="https://user-images.githubusercontent.com/28451557/150168460-ebdf8dcc-5556-4b7a-9c18-52ee7f602c91.png">

**Reply by adkinsrs** [reply]
I think the "tooltip" JS script probably got removed from the HTML

**Reply by adkinsrs** [reply]
Just pushed a commit to get tooltips working again.  I also made a couple of stylistic fixes too.  Unsure when the tooltip functionality actually broke since they did previously work

**Comment by JPReceveur** [comment]
PCA step in single cell wb:

The PCA step tripped up a couple of the users. I think we need to either add a default so the plotting can go through without an error or make it clear in the box that at least one gene is required. 

<img width="1063" alt="Screen Shot 2022-01-19 at 10 49 10 AM" src="https://user-images.githubusercontent.com/28451557/150166164-c800bf71-a6ae-4fc8-a754-721d53a42a61.png">

Also I think we should include an example for the Principle Components option, maybe including e.g. 1,2 so users can see what the input format looks like? Had two users get stuck here and were not sure of what to put into the box.

<img width="1063" alt="Screen Shot 2022-01-19 at 10 49 10 AM" src="https://user-images.githubusercontent.com/28451557/150166164-c800bf71-a6ae-4fc8-a754-721d53a42a61.png">

**Reply by adkinsrs** [reply]
I recently had fixed the issue where the CGI script would fail if you plotted without passing in a gene... now you can do that.

I'll add an example for the PC option soon

**Reply by adkinsrs** [reply]
I added an example for the Principle Component step, and converted the text into a tooltip

**Comment by JPReceveur** [comment]
tSNE/UMAP step in single cell wb:

Provide examples or more info buttons for N neighbors and N dimensions buttons.

Have 'whats the difference' link open in new window, as is it navigates you away from current page

**Reply by adkinsrs** [reply]
9041064d245016358bd7d9b3f23c10d597526f75. addresses the new tab navigation for "what's the difference"

**Comment by JPReceveur** [comment]
Find marker genes step in single cell wb:

Include labels or description for table (i.e. have it written somewhere that columns are clusters and rows are genes)

Is it the convention to number cell clusters starting at 0? Or is this a result of being zero indexed?

When choosing 10 marker genes, 11 are returned.

<img width="640" alt="Screen Shot 2022-01-19 at 11 17 05 AM" src="https://user-images.githubusercontent.com/28451557/150171213-8d2b2314-cb02-4742-bb5c-0c66b50d87b8.png">

**Reply by adkinsrs** [reply]
I just pushed a commit for this.  Turns out the function to grab the top n genes in Pandas was selecting with .loc instead of .iloc . The former includes the end-number and the latter doesn't (like a Python slice), so if N Genes = 5, then .loc would grab rows 0-5 and .iloc would grab rows 0-4.

**Reply by adkinsrs** [reply]
This should be available for testing on production

**Comment by JPReceveur** [comment]
Note on overall layout of homepage:

If you have the webpage shrunk so you can have two windows side by side, the profile option disappears from easy view. 

<img width="941" alt="Screen Shot 2022-01-19 at 11 21 28 AM" src="https://user-images.githubusercontent.com/28451557/150172098-dba1caa4-b013-4891-af8d-c60a5e9c1090.png">

**Comment by JPReceveur** [comment]
Assorted homepage suggestions/comments  from users:
- Popover for different tools to provide more information on what each tool is for
- Had a hard time finding EpiViz displays after being moved into datasets, highlight that they have been moved from previous location
-Wording for selecting profile default button is ambigious, potentially change to make this profile default on home page

<img width="268" alt="Screen Shot 2022-01-19 at 11 35 34 AM" src="https://user-images.githubusercontent.com/28451557/150174289-638f933d-7a10-48e4-bca1-646621326789.png">

- Tutorial/ documentation provides useful info but need to be updated in some cases
- Would like more information somewhere on how the default/ public profiles are selected/curated
- There was some slight confusion between profile on the homepage and 'user profile' 
- Would like some more information on what curation does (during uploading of expression data)

**Reply by adkinsrs** [reply]
<img width="673" alt="Screen Shot 2022-02-03 at 2 09 15 PM" src="https://user-images.githubusercontent.com/5665914/152412304-a92b53f5-019f-429a-9f8e-f15bcb43c9aa.png">

@JPReceveur This does not solve the next-to-last bullet point, but do you think changing the front page gene symbol and profile selection items to start with verbs helps with some confusion?  Was anyone getting confused with the purpose of these fields?

**Reply by JPReceveur** [reply]
I think 'Select Profile' does help a bit and might help people realize that its a button that they can interact with. It also works better with the overall theme of verbs. 

There was some confusion but I think it was less about the layout and more about 'How do we know what goes into a profile?'
A couple people were looking around for an 'all' option but I don't think that something we want to implement.

Would it be worth changing the 'Hearing (default)' text to the same white color as Select profile? All the other text in other purple boxes on the page is bright white

**Reply by adkinsrs** [reply]
I can definitely change the color scheme to match up and I agree with that.  I believe the "purple" color was a holder from the previous front page layout.  As for the other points in this comment (and the discussion thread without a "party popper" emoji), they either haven't been addressed or they are @jorvis territory in my opinion.

**Reply by JPReceveur** [reply]
Sounds good, if you are matching color, want to match font with the other button text as well (text in other buttons looks bolded)? At least for the 'select profile' part

**Reply by adkinsrs** [reply]
<img width="1346" alt="Screen Shot 2022-02-04 at 10 38 29 AM" src="https://user-images.githubusercontent.com/5665914/152557532-b53aaf37-f952-4167-94ff-5424985529f3.png">


This is the styling I have committed to Github

**Comment by JPReceveur** [comment]
Manual/Documentation:
Currently none of the documentation pages have text based documentation. Are we planning on adding this soon or would it be better to remove the 'text based documentation header' until we have something to put there?

Rename Workbench tab in documentation to Single Cell Workbench to help users find it quicker

**Reply by adkinsrs** [reply]
I thought Elena was working on this before she left but I could be wrong.  Perhaps @songeric1107 could clarify.
